frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projectimport/WsImportPortlet.java

```diff
@@ -0,0 +1,160 @@
+
+import javax.portlet.*;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Arrays;
+import java.util.List;
+
+import static com.google.common.base.Strings.nullToEmpty;
+
+/**
+ * @author ksoranko@verifa.io
+ */
+
+public class WsImportPortlet extends Sw360Portlet {
+    private static final Logger LOGGER = Logger.getLogger(WsImportPortlet.class);
+    private static ProjectImportService.Iface projectImportClient = new ThriftClients().makeWsImportClient();
+
+    @Override
+    public void doView(RenderRequest request, RenderResponse response) throws IOException, PortletException {
+        super.doView(request, response);
+    }
+
+    private TokenCredentials getTokenCredentialsFromSession(PortletSession session) {
+        String token = (String) session.getAttribute(ProjectImportConstants.TOKEN);
+        String server = (String) session.getAttribute(ProjectImportConstants.SERVER_URL);
+        String userKey = (String) session.getAttribute(ProjectImportConstants.USER_KEY);
+        return new TokenCredentials()
+                .setToken(token)
+                .setServerUrl(server)
+                .setUserKey(userKey);
+    }
+
+    private void putTokenCredentialsIntoSession(PortletSession session, TokenCredentials tokenCredentials) {
+        session.setAttribute(ProjectImportConstants.TOKEN, nullToEmpty(tokenCredentials.getToken()));
+        session.setAttribute(ProjectImportConstants.SERVER_URL, nullToEmpty(tokenCredentials.getServerUrl()));
+        session.setAttribute(ProjectImportConstants.USER_KEY, nullToEmpty(tokenCredentials.getUserKey()));
+    }
+
+    private List<String> getProjectTokensForImport(PortletRequest request) throws PortletException, IOException {
+        String[] checked = request.getParameterValues("checked[]");
+        List<String> checkedTokens = Arrays.asList(checked);
+
+        return checkedTokens;
+    }
+
+    private boolean isImportSuccessful(ImportStatus importStatus) {
+        return (importStatus.isSetRequestStatus() && importStatus.getRequestStatus().equals(RequestStatus.SUCCESS) && importStatus.getFailedIds().isEmpty());
+    }
+
+    private ImportStatus importDatasources(List<String> toImport, User user, TokenCredentials tokenCredentials) {
+        ImportStatus importStatus = new ImportStatus();
+        try {
+            importStatus = projectImportClient.importData(toImport, user, tokenCredentials);
+            if (!isImportSuccessful(importStatus)) {
+                if (importStatus.getRequestStatus().equals(RequestStatus.FAILURE)) {
+                    LOGGER.error("Importing of data sources failed.");
+                } else {
+                    LOGGER.error("Importing has not succeeded for the following IDs: " + importStatus.getFailedIds().toString());
+                }
+            }
+        } catch (TException e) {
+            LOGGER.error("ImportDatasources: Exception ".concat(e.getMessage()));
```

Comment:
Would be better to return the complete exception with stack and cause.

` LOGGER.error("ImportDatasources failed", e);`